### ISSUE TITLE
bug in font_manager.FontManager.score_family()

### ISSUE THREAD

**Issue by chadawagner** [matplotlib/matplotlib#4607]
If a font to be scored is found first in the font alias list, it currently results in a "perfect" score of 0.
For example:
```
font_manager.FontManager().score_family(['myfont','sans-serif'], 'bitstream vera sans')
```
will return 0, and 'myfont' will never be found if it happens to appear later in the list of known fonts.

Commit 314a0cf already attempted to address this issue, but didn't solve it in the case I'm running into.

**Comment by chadawagner** [comment]
The code for the returned score is currently
```
((0.1 * (idx / len(options))) * ((i + 1) / len(families)))
```
and the issue I'm having occurs when idx is 0. The previous commit addressed the right multiplicand, changing `i` to `i+1` so that it would never be 0, but the overall result is still 0 when the left side is 0.

**Comment by mdboom** [comment]
Yeah -- I think changing to (idx + 1) makes sense here.

**Comment by tacaswell** [comment]
@chadawagner Would you like to put in a PR to fix this?

**Comment by chadawagner** [comment]
I could make a PR, but I'm still not sure the right way to fix this scoring algorithm. I think it's broken in more ways than just changing to `(idx + 1)` would fix.

For example, assume I have a font family list of `['font1', 'font2', 'sans-serif']`, then any font that is part of the sans-serif alias will score at most 0.1, which is lower (better) than font2.

**Comment by chadawagner** [comment]
How about
```
i / len(families) + idx / (len(options) * len(families))
```
That way the returned score is always in the half-open interval `[ i/len(families), (i+1)/len(families) )`

**Comment by mdboom** [comment]
Fixed by #4689
